pkg/event/sqlite.go

```diff
@@ -0,0 +1,300 @@
+}
+
+func (repo SQLiteRepository) ListResourceEvents(ctx context.Context, listParams ListResourceEventParams) ([]ResourceEventModel, string, error) {
+	models := make([]ResourceEventModel, 0)
+	resourceEvents := make([]ResourceEvent, 0)
+	query := `
+		SELECT id, type, source, resourceType, resourceId, meta, createdAt
+		FROM resourceEvent
+		WHERE
+	`
+	conditions := []string{}
+	replacements := []interface{}{}
+
+	if listParams.Type != "" {
+		conditions = append(conditions, "type = ?")
+		replacements = append(replacements, listParams.Type)
+	}
+
+	if listParams.Source != "" {
+		conditions = append(conditions, "source = ?")
+		replacements = append(replacements, listParams.Source)
+	}
+
+	if listParams.ResourceType != "" {
+		conditions = append(conditions, "resourceType = ?")
+		replacements = append(replacements, listParams.ResourceType)
+	}
+
+	if listParams.ResourceId != "" {
+		conditions = append(conditions, "resourceId = ?")
+		replacements = append(replacements, listParams.ResourceId)
+	}
+
+	if listParams.LastId != "" {
+		lastIdSpec, err := stringToLastIdSpec(listParams.LastId)
+		if err != nil {
+			return models, "", service.NewInvalidParameterError("lastId", "")
+		}
+
+		conditions = append(conditions, "(createdAt, id) < (?, ?)")
+		replacements = append(replacements, lastIdSpec.CreatedAt)
+		replacements = append(replacements, lastIdSpec.ID)
+	}
+
+	conditions = append(conditions, "DATE(createdAt) BETWEEN DATE(?) AND DATE(?)")
+	replacements = append(replacements, listParams.Since)
+	replacements = append(replacements, listParams.Until)
+
+	query = fmt.Sprintf("%s %s ORDER BY createdAt DESC, id DESC LIMIT ?", query, strings.Join(conditions, " AND "))
+	replacements = append(replacements, listParams.Limit)
+	err := repo.DB.SelectContext(
+		ctx,
+		&resourceEvents,
+		query,
+		replacements...,
+	)
+	if err != nil {
+		switch err {
+		case sql.ErrNoRows:
+			return models, "", nil
+		default:
+			return models, "", err
+		}
+	}
+
+	for i := range resourceEvents {
+		models = append(models, &resourceEvents[i])
+	}
+
+	if len(resourceEvents) == 0 || len(resourceEvents) < int(listParams.Limit) {
+		return models, "", nil
+	}
+
+	lastResourceEvent := resourceEvents[len(resourceEvents)-1]
+	lastIdStr, err := lastIdSpecToString(LastIdSpec{
+		ID:        lastResourceEvent.ID,
+		CreatedAt: lastResourceEvent.CreatedAt,
+	})
+	if err != nil {
+		return models, "", err
+	}
+
+	return models, lastIdStr, nil
+}
+
+func (repo SQLiteRepository) TrackAccessEvent(ctx context.Context, accessEvent AccessEventModel) error {
+	return repo.TrackAccessEvents(ctx, []AccessEventModel{accessEvent})
+}
+
+func (repo SQLiteRepository) TrackAccessEvents(ctx context.Context, models []AccessEventModel) error {
+	accessEvents := make([]AccessEvent, 0)
+	for _, model := range models {
+		accessEvents = append(accessEvents, *NewAccessEventFromModel(model))
+	}
+
+	result, err := repo.DB.NamedExecContext(
+		ctx,
+		`
+		   INSERT INTO accessEvent (
+			  type,
```

Comment:
Now that we've merged #71, you can insert the `id` column in this query as well to use the code generated UUID.